### ISSUE TITLE
Annotation or enums for preferred constant values

### ISSUE THREAD

**Issue by elliette** [flutter/flutter#157651]
### Use case

As part of work around the Flutter Property Editor (https://github.com/flutter/devtools/issues/1948), we would like to be able to support Flutter properties like [strokeAlign](https://api.flutter.dev/flutter/painting/BorderSide/strokeAlign.html). Although it's a double, there are three constants defined by Flutter that are preferred (strokeAlignInside, strokeAlignCenter, and strokeAlignOutside). Ideally we would like to find a way to support this in the Property Editor without having to hardcode rules for these special cases. 

There are likely other cases similar to `strokeAlign` that should be modified to best support the Flutter Property Editor. 

### Proposal

A couple options to solve this:

1. By creating an annotation (e.g., `@knownValues`) 

```
BorderSide({
    Color color = const Color(0xFF000000),
    double width = 1.0,
    BorderStyle style = BorderStyle.solid,
    @knownValues([BorderSide.strokeAlignInside, BorderSide.strokeAlignCenter, BorderSide.strokeAlignOutside])
    double strokeAlign = strokeAlignInside})
```

2. Switch to an enum 
 
Another option would be to use an enhanced enum with `double` members. However, in the case of `strokeAlign`, because other values besides -1.0. 0.0. and 1.0 can be used, this is likely not the best solution.

**Comment by elliette** [comment]
FYI @Piinks

**Comment by matanlurey** [comment]
I'd personally want to punt this back to the Dart SDK team, i.e. as an annotation in `package:meta` so that `package:analyzer` (for completion), `package:dartdoc` (for documentation), _and_ the Flutter Devtools could take advantage of the same heuristics.

**Comment by bwilkerson** [comment]
I agree that if such an annotation is introduced that it should be in `package:meta`. (It might not be useful to add it depending on how the discussions around the (poorly named) 'enum-shorthand' feature progress because there might be a different way to express the same information. I would prefer to have the annotation.)

It isn't necessary for it to be declared in `package:meta` in order for the analyzer and other tools to support it (if there are Flutter-specific annotations we should be supporting please let us know). But the stronger argument, in my mind, is that this annotation is not Flutter-specific; it would be equally useful for authors of non-Flutter code.

**Comment by nate-thegrate** [comment]
Potential annotation syntax:

#### in package:meta
```dart
class AutofillSuggestions {
  const AutofillSuggestions(this.suggestionList);
  final List<(AutofillLocation, AutofillValue)> suggestionList;
}

sealed class AutofillLocation {
  /// Any parameter with this type would get the relevant suggestions.
  const factory AutofillLocation.everywhere(Type type) = _Everywhere;

  /// The suggestions apply to the specified constructor parameter.
  const factory AutofillLocation.parameter(Function constructor, String parameterName) = _Parameter;
}

sealed class AutofillValue {
  const factory AutofillValue.staticValuesFrom(Type type) = _StaticValues;

  const factory AutofillValue.list(List list) = _FromList;
}
```

#### colors.dart
```dart
@AutofillSuggestions([
  (AutofillLocation.everywhere(Color), AutofillValue.staticValuesFrom(Colors)),
])
library;
```

#### material.dart
```dart
// Since colors.dart is being exported, any library that imports material
// will get color suggestions from the Colors class.
export 'src/material/colors.dart';

@hideSuggestions // Exports the widgets library but ignores any suggestion annotations.
export 'widgets.dart';
```

**Comment by nate-thegrate** [comment]
> Can you give me an example of a place where we would want to ensure that the server doesn't suggest one of the candidates from the exported library?

Good point, I can't think of one! Maybe it's best to adhere to the "lazy programming" philosophy and avoid adding something like `@hideSuggestions` for now.

<br>

> This approach requires that server look at every library in every depended on (or maybe reachable) package.

I was under the impression that the server is already doing this.

<img src="https://github.com/user-attachments/assets/4c93df89-f4b1-48ee-9f25-55feebc17cd5" alt="every package" width="291px" />

**Comment by bwilkerson** [comment]
Maybe so, though I thought it would have to do more work than it's currently doing. I'll have to double check to know for sure.
